pkg/build/api/types.go

```diff
@@ -76,8 +76,73 @@ type BuildSpec struct {
 	// scheduled in the system, that the build may be active on a node before the
 	// system actively tries to terminate the build; value must be positive integer
 	CompletionDeadlineSeconds *int64
+
+	// Info is a placeholder that can store multiple reasons why a build was
+	// triggered. It should cover for a future implementation of level triggered
+	// builds where you can multiple reasons why a build was triggered.
+	Info []*BuildTriggerInfo
```

Comment:
Ok, I think if we're going to add content to the BuildSpec, then we should rework the types (as suggested by @rhcarvalho i think) to:

```
type CommonSpec struct {
  // existing BuildSpec fields go here
}

type BuildSpec struct {
  CommonSpec
  // new build-only (not buildconfig) specific fields go here, such as the Info field being introduced in this PR
}

type BuildConfigSpec {
  CommonSpec
  Triggers []BuildTriggerPolicy
}

type Build struct {
	Spec BuildSpec
	Status BuildStatus
}

type Build struct {
	Spec BuildSpec
	Status BuildStatus
}

type BuildConfig struct {
	Spec BuildConfigSpec
	Status BuildConfigStatus
}
```

This shouldn't necessitate any actual code changes because by embedding the types, all the field references should stay the same.  And since the field names are the same, it also won't affect the api.

@rhcarvalho @mfojtik what do you think?